Code/Framework/AzToolsFramework/AzToolsFramework/Input/QtEventToAzInputMapper.cpp

```diff
@@ -408,32 +439,76 @@ namespace AzToolsFramework
 
     void QtEventToAzInputMapper::HandleKeyEvent(QKeyEvent* keyEvent)
     {
-        // Ignore key repeat events, they're unrelated to actual physical button presses.
+        const Qt::Key key = static_cast<Qt::Key>(keyEvent->key());
+        const QEvent::Type eventType = keyEvent->type();
+
+        // special handling for text events in edit mode
+        {
+            static Qt::Key lastKey = key;
+            static bool consumeAsText = false;
+
+            QString keyText = keyEvent->text();
+            if (key == Qt::Key_Backspace)
+            {
+                keyText = "\b";
+            }
+
+            if (!keyText.isEmpty())
+            {
+                // key events are first sent as shortcuts, if accepted they are then resent as traditional key
+                // down events.  dispatching the key event as text on shortcut (and auto-repeat press) ensures
+                // all printable keys a fair chance at being consumed before processing elsewhere
+                if (eventType == QEvent::Type::ShortcutOverride ||
+                    (eventType == QEvent::Type::KeyPress && keyEvent->isAutoRepeat()))
+                {
+                    bool textConsumed = false;
+                    AzFramework::InputTextNotificationBus::Broadcast(
+                        &AzFramework::InputTextNotifications::OnInputTextEvent,
+                        AZStd::string(keyText.toUtf8().data()),
+                        textConsumed
+                    );
+                    consumeAsText = textConsumed;
+                }
+                // follow up events from accepting the previous key event also need to be accepted without
+                // dispatching the text event, otherwise the key event will be processed twice: once for the
+                // previous text event, then again as a normal key event which can lead to odd behaviour
+                // during text edit
+                else
+                {
+                    consumeAsText &= (lastKey == key);
```

Comment:
update: Please don't use bitwise operations for boolean tests such as this, this should be

`consumeAsText = consumeAsText && lastKey == key;`

There are some subtle bugs that can creep in when mixing boolean and bitwise operations.

Reply:
Fair point but as mentioned in another comment thread, this logic was extraneous and removed.

Reply:
Okay good to know thanks 👍